tests/quantization/test_quantization_metric.py

```diff
@@ -203,48 +92,7 @@ def test_network_quantization_share_metric(network_quantization_share_metric_tes
         assert qmetric_stat[key] == approx(value, rel=1e-2)
 
 
-MEMORY_COST_METRIC_TEST_CASES = [
-    TestStruct(
-        initializers={},
-        activations={},
-        weights={},
-        ignored_scopes=[],
-        target_device='TRIAL',
-        quantizer_setup_type='pattern_based',
-        table={MemoryCostMetric.EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR: 4.0,
-               MemoryCostMetric.SIZE_MEMORY_FP_WEIGHTS_STR: 88.74,
-               MemoryCostMetric.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR: 22.18,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_COMPRESSED_MODEL_STR: 0.0625,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_FP32_MODEL_STR: 0.0625}),
-    TestStruct(
-        initializers={"precision": {
-            "bitwidth_per_scope":
-            [[2, 'InsertionType.NNCF_MODULE_PRE_OP AlexNet/Sequential[features]/NNCFConv2d[0]'],
-             [4, 'InsertionType.NNCF_MODULE_PRE_OP AlexNet/Sequential[features]/NNCFConv2d[6]']]
-        }},
-        activations={},
-        weights={"bits": 8},
-        ignored_scopes=[],
-        target_device='TRIAL',
-        quantizer_setup_type='pattern_based',
-        table={MemoryCostMetric.EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR: 4.05,
-               MemoryCostMetric.SIZE_MEMORY_FP_WEIGHTS_STR: 88.74,
-               MemoryCostMetric.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR: 21.86,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_COMPRESSED_MODEL_STR: 0.0625,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_FP32_MODEL_STR: 0.0625}),
-    TestStruct(
-        initializers={},
-        activations={},
-        weights={},
-        ignored_scopes=['AlexNet/Sequential[features]/NNCFConv2d[0]'],
-        target_device='TRIAL',
-        quantizer_setup_type='pattern_based',
-        table={MemoryCostMetric.EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR: 3.99,
-               MemoryCostMetric.SIZE_MEMORY_FP_WEIGHTS_STR: 88.74,
-               MemoryCostMetric.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR: 22.19,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_COMPRESSED_MODEL_STR: 0.0625,
-               MemoryCostMetric.MAX_MEMORY_CONSUMPTION_ACTIVATION_TENSOR_IN_FP32_MODEL_STR: 0.0625}),
-]
+MEMORY_COST_METRIC_TEST_CASES = []
```

Comment:
@asenina it seems that all of the memory cost metrics have been written for the pattern-based mode. Please open a separate PR that adds propagation-based test cases ASAP, otherwise after the pattern-based mode removal the memory cost metric will remain untested.

Reply:
Done. #546